### PR TITLE
Issue 14768 - Error: index 174762 overflow for static array

### PR DESCRIPTION
**Pull request by 9rnsr**
https://issues.dlang.org/show_bug.cgi?id=14768

Improve diagnostic message.

### CODE REVIEW

File: src/mtype.c [hunk 1]
```diff
@@ -4067,7 +4067,8 @@ Type *TypeSArray::semantic(Loc loc, Scope *sc)
             if (mulu(tbn->size(loc), d2, overflow) >= 0x1000000 || overflow) // put a 'reasonable' limit on it
             {
               Loverflow:
-                error(loc, "index %llu overflow for static array", (unsigned long long)d1);
+                error(loc, "%s size %llu * %llu exceeds 16MiB size limit for static array",
```
Comment (WalterBright) [inline src/mtype.c]: Mb, not MiB

Comment (9rnsr) [inline src/mtype.c]: Done.

Comment (Geod24) [inline src/mtype.c]: @WalterBright : [Apparently MiB would be more correct here](https://en.wikipedia.org/wiki/Mebibyte).

**Comment by aG0aep6G** [comment]
> Mb, not MiB

Uh, I think that's misleading. We mean mebibyte/megabyte (= 1024*1024 bytes), right?

"Mb" is bad, because: "The megabit has the unit symbol Mb or Mbit." (https://en.wikipedia.org/wiki/Megabit)

"MB" is ok, if imprecise, because it's understood that it can mean 1000*1000 bytes or 1024*1024 bytes. (https://en.wikipedia.org/wiki/Megabyte)

"MiB" is ok, if uncommon, as it means exactly 1024*1024 bytes. (https://en.wikipedia.org/wiki/Mebibyte)

**Comment by WalterBright** [comment]
I stand corrected. Sorry, Kenji!

**Comment by 9rnsr** [comment]
OK, changed back to `MiB`.

**Pull request merged**